### ISSUE TITLE
rand() function inconsistent with other databases

### ISSUE THREAD

**Issue by drstrangelooker** [ClickHouse/ClickHouse#7324]
The rand() function in most databases yields a floating point value between 0 and 1. ClickHouse's rand() yields a value between 0 and 2^32-1. This is an unexpected result to most experienced database users.

**Comment by hczhcz** [comment]
Probably it is better to have `randInt32()`, `randInt64()`, `randFloat32()`, `randFloat64()`, etc.?

**Comment by alexey-milovidov** [comment]
@hczhcz Yes. And for rand() we should have a setting controlling its behaviour for backward compatibility.

**Comment by qoega** [comment]
@alexey-milovidov do you still think we need to change behavior or just add a setting to control it to have compatible behavior for those who need it?

We have randCanonical for 0-1

**Comment by alexey-milovidov** [comment]
We can keep it as it is.
